daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -1503,7 +1718,92 @@ func (devices *DeviceSet) DeleteDevice(hash string) error {
 	devices.Lock()
 	defer devices.Unlock()
 
-	return devices.deleteDevice(info)
+	// If mountcount is not zero, that means devices is still in use
+	// or has not been Put() properly. Fail device deletion.
+
+	if info.mountCount != 0 {
+		return fmt.Errorf("devmapper: Can't delete device %v as it is still mounted. mntCount=%v", info.Hash, info.mountCount)
```

Comment:
This is more of a saftey check to make sure that docker has done proper number of Get()/Put() so that mount count is zero. If not, then it is most likely  a upper layer bug where a Get() was done but corresponding Put() has not been done and higher layer is trying to delete the device. So instead of scheduling a deferred delete now, it will force higher layers to first do all corresponding Put() and then call into delete device logic. If device is busy even now as it leaked into other namespaces, then deferred deletion will be scheduled.